soc/xtensa/esp32s2/CMakeLists.txt

```diff
@@ -3,4 +3,79 @@
 zephyr_sources(
   soc.c
   soc_cache.c
+  loader.c
   )
+
+if(CONFIG_BOOTLOADER_ESP_IDF)
```

Comment:
guess moving code around is accepted :wink:

Reply:
I supposed it is. Besides this, keeping this in board level isn't correct.